### PR TITLE
Improve error reporting of get_satpos

### LINKED ISSUE #1496: Improve error reporting of satpy.utils.get_satpos
## Feature Request

**Is your feature request related to a problem? Please describe.**

Some computations in satpy require the actual satellite position (lat, lon, alt), which can be provided via attributes `orbital_parameters` or `satellite_longitude/latitude/altitude` (deprecated). The helper method `satpy.utils.get_satpos` tries to extract the best match from the available dataset attributes. If none of the above attributes are present, the method fails with a `KeyError('satellte_longitude')`, see #1494 .

**Describe the solution you'd like**
The method to provide a more informative error message. Suggestion: 

```
KeyError("Unable to determine actual satellite position. Either the reader doesn't provide that information or geolocation datasets are not available").
```

**Describe any changes to existing user workflow**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by sfinkens**
Improve error reporting of get_satpos as proposed in #1496 

 - [X] Closes #1496 
 - [X] Tests added

### CODE REVIEW

**Comment by sfinkens** [comment]
@djhoese Thanks for the pointer to that issue. I've simplified the exception handling and added a warning when the orbital parameters are missing. That should address #1907, but let's wait for @gerritholl 's review.

File: satpy/utils.py [hunk 1]
```diff
@@ -284,34 +284,43 @@ def atmospheric_path_length_correction(data, cos_zen, limit=88., max_sza=95.):
...
+        except KeyError:
+            raise KeyError("Unable to determine satellite position. Either"
+                           "the reader doesn't provide that information or"
+                           "geolocation datasets are not available.")
 
+
+def _get_satpos(dataset):
+    orb_params = dataset.attrs['orbital_parameters']
+    alt = _get_sat_altitude(orb_params)
+    lon, lat = _get_sat_lonlat(orb_params)
```
Comment (mraspaud) [inline satpy/utils.py:319]: Is this tested?

**Review by mraspaud — COMMENTED** [review]
Codecov claims that _get_satpos and _get_satpos_legacy isn't tested? otherwise lgtm

**Comment by sfinkens** [comment]
@mraspaud Coverage locally tells me the lines (284-358) are covered

```
$ coverage report --show-missing
Name                                      Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------
satpy/satpy/utils.py     248     42    83%   36-37, 45-47, 103, 154-161, 169-172, 240, 258-281, 369-375, 396, 400-402, 468, 500-502
```

File: satpy/tests/test_utils.py [hunk 2]
```diff
@@ -254,9 +254,28 @@ def test_get_satpos(self, warn_mock):
...
+        {'satellite_altitude': 1}
+    )
+)
+def test_get_satpos_fails_with_informative_error(attrs):
+    """Test that get_satpos raises an informative error message."""
+    dataset = mock.MagicMock(attrs=attrs)
+    with pytest.raises(KeyError) as err:
+        get_satpos(dataset)
+    assert not str(err.value).startswith("satellite_")
+    assert not str(err.value).startswith("orbital_")
```
Comment (gerritholl) [inline satpy/tests/test_utils.py:278]: Alternately, you could use `pytest.raises(KeyError, match="...")` here to test for an exact match of the expected error message.

**Review by gerritholl — COMMENTED** [review]
I'm confused by codecov messages about missing test coverage, because from reading the source code it would seem these should be tested.  Could you either confirm those are in fact tested and codecov is wrong, or check why they might not be tested at all?  Maybe something is inadvertently mocked?

**Comment by djhoese** [comment]
I completely didn't see this PR before working on my #2030

**Comment by sfinkens** [comment]
Thanks @gerritholl for your review! I guess this is kind of superseded by #2030 and hopefully we can include the relevant bits in #2030 :slightly_smiling_face:

File: satpy/tests/test_utils.py (hunk 2 above)
Comment (djhoese) [reply satpy/tests/test_utils.py:278]: I'm moving this to my PR and these don't actually work as intended. The KeyError returned includes a single quote around the missing key so `startswith` never matches it. I'll try using the match regex.

**Pull request closed without merging**